data_steward/cdr_cleaner/cleaning_rules/truncate_fitbit_data.py

```diff
@@ -0,0 +1,213 @@
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+CUTOFF_DATE = '2019-11-26'
+CUTOFF_DATETIME = '2019-11-26T00:00:00'
+
+FITBIT_DATE_TABLES = [common.ACTIVITY_SUMMARY, common.HEART_RATE_SUMMARY]
+FITBIT_DATETIME_TABLES = [common.HEART_RATE_MINUTE_LEVEL, common.STEPS_INTRADAY]
+
+FITBIT_TABLES_DATE_FIELDS = {
+    common.ACTIVITY_SUMMARY: 'date',
+    common.HEART_RATE_SUMMARY: 'date'
+}
+FITBIT_TABLES_DATETIME_FIELDS = {
+    common.HEART_RATE_MINUTE_LEVEL: 'datetime',
+    common.STEPS_INTRADAY: 'datetime'
+}
+
+# Save rows that will be dropped to a sandboxed dataset
+SANDBOX_QUERY = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox}}.{{intermediary_table}}` AS (
+SELECT * 
+FROM `{{project}}.{{dataset}}.{{table_name}}`
+WHERE {{date_field}} < '{{cutoff_date}}')""")
+
+# Drop any FitBit data that is newer than the cutoff date
+TRUNCATE_FITBIT_DATA_QUERY = jinja_env.from_string("""
+SELECT * FROM `{{project}}.{{dataset}}.{{table_name}}` t
+WHERE person_id NOT IN (
+SELECT person_id FROM `{{project}}.{{sandbox}}.{{intermediary_table}}`)""")
+
+
+class TruncateFitbitData(BaseCleaningRule):
+    """
+    All rows of FitBit data with dates before 11/26/2019 should be moved from the Activity Summary,
+    Heart Rate Minute Level, Heart Rate Summary, and Steps Intraday tables to a sandboxed FitBit dataset
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'All rows of data in the FitBit tables with dates before 11/26/2019 will be truncated.'
+        super().__init__(issue_numbers=['DC1046'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.FITBIT],
+                         affected_tables=common.FITBIT_TABLES,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        date_table_counter = 0
+        datetime_table_counter = 1
+        sandbox_queries = []
+        truncate_queries = []
+
+        # Sandboxes and truncates data from FitBit tables with date
+        for table in FITBIT_DATE_TABLES:
```

Comment:
Instead of doing the counter's you can always do the following:
`for i, table in enumerate(FITBIT_DATE_TABLES)`
.....
`self.get_sandbox_tablenames()[i]`

Reply:
I'm using the counters here because I need to specifically access elements in the sandbox table names list. Is there a way to use enumerate and still access exactly want I need?

Reply:
Okay that's what I though, I'll play around with how my `sandbox_tablenames` are generated to see if I can use enumerate because it will make the code a bit cleaner!